cdc/log.cc

```diff
@@ -68,6 +69,70 @@ namespace cdc {
 static schema_ptr create_log_schema(const schema&, std::optional<utils::UUID> = {});
 }
 
+static constexpr auto cdc_group_name = "cdc";
+
+void cdc::stats::parts_touched_stats::register_metrics(seastar::metrics::metric_groups& metrics, sstring suffix) {
+    namespace sm = seastar::metrics;
+    auto register_part = [&] (part_type part, sm::label_instance part_instance) {
+        metrics.add_group(cdc_group_name, {
+                sm::make_total_operations("components_touched_" + suffix, count[(size_t)part],
+                        sm::description("number of times a mutation part type have been touched during cdc mutation augmentation"),
+                        {part_instance})
+            });
+    };
+
+    const auto part_label = sm::label("part");
+    register_part(part_type::STATIC_ROW, part_label("static_row"));
+    register_part(part_type::CLUSTERING_ROW, part_label("clustering_row"));
+    register_part(part_type::MAP, part_label("map"));
+    register_part(part_type::SET, part_label("set"));
+    register_part(part_type::LIST, part_label("list"));
+    register_part(part_type::UDT, part_label("udt"));
+    register_part(part_type::RANGE_TOMBSTONE, part_label("range_tombstone"));
+    register_part(part_type::PARTITION_DELETE, part_label("partition_delete"));
+    register_part(part_type::ROW_DELETE, part_label("row_delete"));
+}
```

Comment:
I used labels here to differentiate counters that track the "touches" of different parts of mutation. I admit that the value they sum up to doesn't have much sense in itself, so I'm open to ideas on how to improve the representation of those counters.